libs/accounts/rate-limit/src/lib/util.ts

```diff
@@ -35,3 +40,85 @@ export function calculateRetryAfter(now: number, block: BlockRecord) {
 export function sanitizeKeyValue(val: string) {
   return val.replace(':', ' ');
 }
+
+export function getLargestRetryAfter(blocks: BlockStatus[]) {
+  if (blocks.length > 0) {
+    let ban = blocks[0];
+    blocks.forEach((x) => {
+      if (x.retryAfter > ban.retryAfter) {
+        ban = x;
+      }
+    });
+    return ban;
```

Comment:
small style suggestion: `return Math.max(...blocks.map(x => x.retryAfter))`?  A little easier to read for me but that's a personal preference of course.

Reply:
I like it.